linkedList.js

```diff
@@ -43,21 +43,19 @@ function linkedListGenerator(){
         },
         remove: function(index){
             let current = this.get(index);
-            let before, after;
+            let before;
             if (this.get(index) === false){
                 return false;
             }
             if (index > 0){
                 before = this.get(index - 1);
+                if (!this.get(index + 1)){
+                    tail = before;
+                }
             } else {
```

Comment:
The else still has the same issue as before. Try to think of remove as three cases, removing at 0, removing in the middle and removing from the end.